### ISSUE TITLE
Access token authentication without a database

### ISSUE THREAD

**Issue by rvishna** [jobec/django-auth-adfs#106]
Thank you for this really useful package! I use it in my DRF web API to authenticate users against Azure AD and it works like a charm.

I'm trying to use the access token backend without a database. My app doesn't need to store anything in the database (I don't use the ORM at all). When I'm using the access token backend, I notice that it creates a new user in the auth_user table. I'm not sure why this is necessary (besides some metadata like the last login time of the user), as the authentication works even if I delete the user. Is it possible to authenticate the user based on claims and not create a user? Perhaps if I use an anonymous user model?

**Comment by JonasKs** [comment]
That’s implemented in #96. 

Set `CREATE_NEW_USERS` to False.

**Comment by JonasKs** [comment]
Oh, it's not pushed to PyPi it seems. @jobec, could you do that?

In the meantime, you can install the latest commit from git by using this command:  
```bash
pip install git+git://github.com/jobec/django-auth-adfs@41fa2b8eca8f80be9db2026a6310673f9cbde1bb
```

**Comment by rvishna** [comment]
This is a useful feature, but not quite what I was asking about. If the user doesn't already exist, this raises a PermissionDenied exception. My question was is it possible to avoid writing anything to the database altogether. If I'm only interested in authentication and don't care about permissions and groups, can I use the logic from this package that verifies the claims and authenticates the user without ever touching the database?

I understand this is a niche use case - I'm trying to see if I can make my DRF  web API completely stateless. I don't think it is possible to remove django.contrib.auth from INSTALLED_APPS and still support user authentication, but I could be wrong.

**Comment by JonasKs** [comment]
Aah, I see. I was a bit quick on that one, sorry. Yeah, that’s not possible with the current implementation. 

Not sure about the last without testing it my self, honestly.

**Comment by JonasKs** [comment]
Did you end up finding a solution, @rvishna ?

**Comment by rvishna** [comment]
@JonasKs I don't think it's possible to authenticate a user without a database. I tried to implement a custom User model which overrides the save and get_or_create method (in the manager class), but this still doesn't circumvent the models for the contenttypes app. I guess if the groups claim is set to None, it is possible to implement a custom User model that inherits from AbstractBaseUser (imported from django.contrib.auth.base_user, instead of django.contrib.auth.models). Then, it might be possible to get away with removing django.contrib.contenttypes and django.contrib.auth from the INSTALLED_APPS - which would result in python manage.py migrate not creating any tables at all.

However.... I never got this far. I realized that if I weren't using a database and not utilizing the ORM or serializers capabilities of the DRF, I might as well use flask. I suppose it may be possible to use the authentication logic in the Access Token backend in a middleware and avoid calling the authenticate() method altogether. If someone wants to make a feature request to this end, I will reference this issue #. Otherwise, I think it's best to close this for now.
